src/nbt/compound.rs

```diff
@@ -2,59 +2,58 @@ use bytes::{Buf, BufMut, Bytes, BytesMut};
 use crab_nbt::nbt::tag::NbtTag;
 use crab_nbt::nbt::utils::{get_nbt_string, END_ID};
 use derive_more::Into;
-use std::collections::HashMap;
+use std::collections::{hash_map::IntoIter, HashMap};
 
-#[derive(Debug, PartialEq, Clone, Default, Into)]
+use crate::error::Error;
+
+/// Compound of named nbt tags.
+#[derive(Clone, PartialEq, Debug, Default, Into)]
 pub struct NbtCompound {
     pub child_tags: HashMap<String, NbtTag>,
 }
 
 impl NbtCompound {
-    pub fn new(child_tags: HashMap<String, NbtTag>) -> NbtCompound {
-        NbtCompound { child_tags }
+    /// Creates new, empty compound
+    pub fn new() -> NbtCompound {
+        NbtCompound {
+            child_tags: HashMap::new(),
+        }
     }
 
-    pub(crate) fn deserialize(bytes: &mut Bytes) -> NbtCompound {
-        let mut compound_tags = HashMap::new();
+    /// Reads compound data from bytes
+    pub fn deserialize_raw(bytes: &mut Bytes) -> Result<NbtCompound, Error> {
```

Comment:
Why this is `raw`? You serialize NbtCompound into bytes. There are no other variants.

Reply:
I wanted to make it consitent with naming in NbtTag, which works like this
raw - only the data
not raw - type id + data

Reply:
Ymmm, I understand but I still think that we should change it to just `deserialize`.

Reply:
Isn't it going to be confusing to the users of the library that methods with identical names work differently?

Reply:
It will be confusing why compound has only `serialize_raw`, why it is raw?

Reply:
> It will be confusing why compound has only `serialize_raw`

Ymmm, because there is no good use for having `serialize`? But we could add it if you want.
> why it is raw?

It's raw because it serializes only data.

Btw. maybe we could rename `serialize_raw` methods to `serialize_data` so the name is more clear on what it does? Because raw is a bit meaningless without a context.